### ISSUE TITLE
Link issues with v 0.6.1

### ISSUE THREAD

**Issue by andersfugmann** [mirage/digestif#44]
My dune project uses digestif and just adds `digestif.c`to the list of libraries:
`(libraries digestif.c ....)`
 
\- and it works. However with version 0.6.1 inline tests no longer link, resulting in the error: 
```
Error: Files /home/afu/.opam/4.06.1/lib/digestif/c/digestif_c.cmxa
       and /home/afu/.opam/4.06.1/lib/digestif/rakia/rakia.cmxa
       both define a module named Rakia
```
building regular binaries still works with 0.6.1 
Switching to digestif.ocaml` does solve the linking problem. 

Could it be that version 0.6.1 requires different linking? I see some mention in the release notes for version 0.6.1.

**Comment by dinosaure** [comment]
I'm currently in vacation for one week. When I come back, I will see what is the problem exactly.

**Comment by samoht** [comment]
You should probably link with `digestif` in your libraries, and `digestif.ocaml` / `digestif.c` only in your final binaries/test executables.

**Comment by andersfugmann** [comment]
Changing to link with `digestif` in libraries and link with `digestif.c` in the executable stanzas does not solve the problem:

```
Error: No implementations provided for the following modules:
         Digestif referenced from aws-s3/aws_s3.cmxa(Aws_s3__Authorization),
           aws-s3/aws_s3.cmxa(Aws_s3__Aws)
```
I guess this has to do with the libraries having a public name in dune. 

Also, since I'm building a library its slightly problematic to require users to add digestif.{c|ocaml} in addition to this library. I'd rather use dune to choose digestif.c if present and fallback to digestif.ocaml in my library, and not burden users of my library with this (Assuming that the installer will only install digestif.c for systems that supports it).

**Comment by andersfugmann** [comment]
Thanks for the prompt relies. 

Have you considered splitting up digestif into two distinct opam packages, i.e. `digestif.c`and digestif.ocaml.` I think it would be rather easy to instruct dune to select the one installed (or choose the c version if present) for projects with direct dependencies on digestif. This would avoid the 

For now I will pin the digestif library version to 0.6, as I cannot get 0.6.1 to work in its current state and I like to avoid transitive dependencies.

**Comment by dinosaure** [comment]
Partially fixed by #53 but the issue about `dune` (and transitivity of `digestif.{c,ocaml}`) is not fixed.

**Comment by dinosaure** [comment]
So it will be fixed now with `dune` (I talked with @rgrinberg and @diml about that) and it's outside the scope of `digestif`. I close this issue.
